cmd/site-replication.go

```diff
@@ -352,54 +392,80 @@ func (c *SiteReplicationSys) AddPeerClusters(ctx context.Context, sites []madmin
 
 	// Create a local service account.
 
-	// Generate a secret key for the service account.
+	// Generate a secret key for the service account if not created already.
 	var secretKey string
-	_, secretKey, err := auth.GenerateCredentials()
-	if err != nil {
-		return madmin.ReplicateAddStatus{}, SRError{
-			Cause: err,
-			Code:  ErrInternalError,
+	svcCred, _, err := globalIAMSys.getServiceAccount(ctx, siteReplicatorSvcAcc)
+	switch {
+	case err == errNoSuchServiceAccount:
+		_, secretKey, err = auth.GenerateCredentials()
+		if err != nil {
+			return madmin.ReplicateAddStatus{}, SRError{
+				Cause: err,
+				Code:  ErrInternalError,
+			}
 		}
-	}
-
-	svcCred, err := globalIAMSys.NewServiceAccount(ctx, sites[selfIdx].AccessKey, nil, newServiceAccountOpts{
-		accessKey: siteReplicatorSvcAcc,
-		secretKey: secretKey,
-	})
-	if err != nil {
-		return madmin.ReplicateAddStatus{}, errSRServiceAccount(fmt.Errorf("unable to create local service account: %w", err))
-	}
-
-	// Notify all other Minio peers to reload user the service account
-	if !globalIAMSys.HasWatcher() {
-		for _, nerr := range globalNotificationSys.LoadServiceAccount(svcCred.AccessKey) {
-			if nerr.Err != nil {
-				logger.GetReqInfo(ctx).SetTags("peerAddress", nerr.Host.String())
-				logger.LogIf(ctx, nerr.Err)
+		svcCred, err = globalIAMSys.NewServiceAccount(ctx, sites[selfIdx].AccessKey, nil, newServiceAccountOpts{
+			accessKey: siteReplicatorSvcAcc,
+			secretKey: secretKey,
+		})
+		if err != nil {
+			return madmin.ReplicateAddStatus{}, errSRServiceAccount(fmt.Errorf("unable to create local service account: %w", err))
+		}
+		// Notify all other Minio peers to reload user the service account
+		if !globalIAMSys.HasWatcher() {
+			for _, nerr := range globalNotificationSys.LoadServiceAccount(svcCred.AccessKey) {
+				if nerr.Err != nil {
+					logger.GetReqInfo(ctx).SetTags("peerAddress", nerr.Host.String())
+					logger.LogIf(ctx, nerr.Err)
+				}
 			}
 		}
+	case err == nil:
+		secretKey = svcCred.SecretKey
+	default:
+		return madmin.ReplicateAddStatus{}, errSRBackendIssue(err)
 	}
 
 	joinReq := madmin.SRInternalJoinReq{
 		SvcAcctAccessKey: svcCred.AccessKey,
-		SvcAcctSecretKey: svcCred.SecretKey,
+		SvcAcctSecretKey: secretKey,
 		Peers:            make(map[string]madmin.PeerInfo),
 	}
-	for i, v := range sites {
-		joinReq.Peers[deploymentIDs[i]] = madmin.PeerInfo{
+
+	if c.enabled {
+		for _, v := range currSites.Sites {
+			joinReq.Peers[v.DeploymentID] = madmin.PeerInfo{
+				Endpoint:     v.Endpoint,
+				Name:         v.Name,
+				DeploymentID: v.DeploymentID,
+			}
+		}
+	}
+
```

Comment:
Do we need the block above? Looks like `sites` would be a superset of `currSites.Sites`, so the block below is simply overwriting with the same values.

Reply:
removed this block